expression/integration_test.go

```diff
@@ -7378,3 +7378,109 @@ func (s *testIntegrationSuite) TestIssue11645(c *C) {
 	tk.MustQuery(`SELECT DATE_ADD('0001-01-02 00:00:00', INTERVAL -25 HOUR);`).Check(testkit.Rows("0000-00-00 23:00:00"))
 	tk.MustQuery(`SELECT DATE_ADD('0001-01-02 00:00:00', INTERVAL -8785 HOUR);`).Check(testkit.Rows("0000-00-00 23:00:00"))
 }
+
+func (s *testIntegrationSuite) TestIssue19892(c *C) {
+	defer s.cleanEnv(c)
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("USE test")
+	tk.MustExec("CREATE TABLE dd(a date, b datetime)")
+
+	// check NO_ZERO_DATE
+	{
+		tk.MustExec("SET sql_mode=''")
+		{
+			tk.MustExec("TRUNCATE TABLE dd")
+			tk.MustExec("INSERT INTO dd(a) values('0000-00-00')")
+			tk.MustQuery("SHOW WARNINGS").Check(testkit.Rows())
+			tk.MustQuery("SELECT a FROM dd").Check(testkit.Rows("0000-00-00"))
+
+			tk.MustExec("TRUNCATE TABLE dd")
+			tk.MustExec("INSERT INTO dd(b) values('2000-10-01')")
+			tk.MustExec("UPDATE dd SET b = '0000-00-00'")
+			tk.MustQuery("SHOW WARNINGS").Check(testkit.Rows())
+			tk.MustQuery("SELECT b FROM dd").Check(testkit.Rows("0000-00-00 00:00:00"))
+		}
+
+		tk.MustExec("SET sql_mode='NO_ZERO_DATE'")
+		{
+			tk.MustExec("TRUNCATE TABLE dd")
+			tk.MustExec("INSERT INTO dd(b) values('0000-00-00')")
+			tk.MustQuery("SHOW WARNINGS").Check(testkit.Rows("Warning 1292 Incorrect datetime value: '0000-00-00 00:00:00'"))
+			tk.MustQuery("SELECT b FROM dd").Check(testkit.Rows("0000-00-00 00:00:00"))
+
+			tk.MustExec("TRUNCATE TABLE dd")
+			tk.MustExec("INSERT INTO dd(a) values('2000-10-01')")
+			tk.MustExec("UPDATE dd SET a = '0000-00-00'")
+			// tk.MustQuery("SHOW WARNINGS").Check(testkit.Rows("Warning 1292 Incorrect date value: '0000-00-00'"))
```

Comment:
Could you tell me why comment it here？

Reply:
Seems there was two identical warning messages , but I have no idea why that happened

Reply:
maybe it's to check insert and update both generating a warning

Reply:
But the insert stmt is valid, no warnings should generated

Reply:
> But the insert stmt is valid, no warnings should generated

i mean line 7521

Reply:
line 7521

Reply:
i mean line 7521 and 7527 should both be checked

Reply:
thanks for the advice, and is there any way to clear warnings before an stmt executing.

Reply:
Seems the warning messages are automatically deleted after a valid STMT executed, IMO, line 7521's warning will not affect this line.

Reply:
seems our communication is misleading 🤣 i mean this line shouldn't be commented as it could check above update generated a warning. and maybe you say that it generated two identical warnings? that's an unexpected behaviour, maybe some bug in code

Reply:
:joy: sorry for the misunderstanding. In my local tests, the warning count maybe 1 or 2. currently, rebase from the master branch and remove the comment, the local test seems passed.  Push it up to see the CI passed or not.

Reply:
It's passed 🎉